src/components/Steps/StudentDetails.tsx

```diff
@@ -1,107 +1,61 @@
-import React, { Dispatch, SetStateAction, useState } from "react";
-import Select from "react-select";
-import styles from "../../styles/Home.module.css";
+import { Step } from "@/pages/SessionCreator";
+import styles from "@/styles/Home.module.css";
+import { MyForm } from "@/types/FormTypes";
+import { ActiveFormProps } from "@/types/types";
+import { SubmitHandler, useForm } from "react-hook-form";
 import {
+  BatchOptions,
   CourseOptions,
   GradeOptions,
   ProgramOptions,
   StreamOptions,
-  BatchOptions,
 } from "../Options/StudentDetailsOptions";
-import { OptionType } from "../Options/StudentDetailsOptions";
+import SelectField from "./Form/SelectedField";
 
-const Steps = {
-  StudentDetails: "StudentDetails",
-  TestDetails: "TestDetails",
-};
-// Renders the sub-page containing student details
 export default function StudentDetails({
+  data,
   setActiveStep,
-}: {
-  setActiveStep: Dispatch<SetStateAction<string>>;
-}) {
-  const [selectedProgram, setSelectedProgram] = useState<OptionType | null>(
-    ProgramOptions[0]
-  );
-  const [selectedBatch, setSelectedBatch] = useState<OptionType | null>(
-    BatchOptions[0]
-  );
+  setData,
+}: ActiveFormProps) {
+  const { register, handleSubmit, control } = useForm<MyForm>({
+    defaultValues: { ...data.student },
+  });
 
-  const [selectedGrade, setSelectedGrade] = useState<OptionType | null>(
-    GradeOptions[0]
-  );
-  const [selectedCourse, setSelectedCourse] = useState<OptionType | null>(
-    CourseOptions[0]
-  );
-  const [selectedStream, setSelectedStream] = useState<OptionType | null>(
-    StreamOptions[0]
-  );
+  const onSubmit: SubmitHandler<MyForm> = (student) => {
```

Comment:
right now i can see that you are using `onSubmit` to build the final form object.. just ensure that each partial submit doesn't go to backend / API-call in next PR.

Reply:
actually maybe we could save at each step too. sort of like a draft form.

Reply:
but a draft form makes sense only if we have login / auth / user handling

Reply:
We are managing the state and not storing them as of now , would make it slow if we tried to save each form and then the user went back and edited that